utils/check-dependencies-exist.js

```diff
@@ -42,52 +43,94 @@ const labelMap = {
-                );
-              })
-              .join('\n');
-          }
+/**
+ * @typedef {{
+ *   dependencies?: string[],
+ *   devDependencies?: string[],
+ *   missing?: { [key: string]: string[] },
+ * }} DependencyErrors
+ */
+
+/**
+ * @param {string} pkgFile
+ * @param {DependencyErrors} errors
+ */
+function formatErrorMessage(pkgFile, errors) {
+  return Object.keys(errors)
+    .map((/** @type keyof DependencyErrors */ key) => {
+      if (errors[key].constructor === Object) {
+        if (key.startsWith('invalid')) {
           return Object.keys(errors[key])
-            .map(pkg =>
-              errors[key][pkg]
-                .map(
-                  errorFile =>
-                    `${labelMap[key].color(labelMap[key].label.padEnd(20))}  ${colors.cyan(
-                      pkg
-                    )}  ${colors.dim(errorFile)}`
-                )
-                .join('\n')
-            )
-            .join('\n');
-        } else {
-          return errors[key]
-            .map(
-              error =>
-                `${labelMap[key].color(labelMap[key].label.padEnd(20))}  ${colors.cyan(error)}`
-            )
+            .map(file => {
+              const line = errors[key][file].line;
+              const column = errors[key][file].column;
+              const message = errors[key][file].formatted;
+
+              return (
+                labelMap[key].color(labelMap[key].label.padEnd(20)) +
+                colors.dim(`  ${line}:${column}  `) +
+                message
+              );
+            })
             .join('\n');
         }
-      })
-      .join('\n')
-  );
+        return Object.keys(errors[key])
+          .map(packageName => {
+            return errors[key][packageName]
+              .map(errorFile => {
+                const contents = fs.readFileSync(errorFile).toString();
+                const line = findLineInFile(contents, packageName);
+                const char = findCharacterInFile(contents, packageName);
+
+                return `${errorFile}\n  ${colors.dim(`${line}:${char}`)}  ${colors.red(
+                  'error'
+                )}  ${colors.yellow(packageName)} is missing from ${pkgFile}  ${colors.dim(
+                  'check-dependencies-exist'
+                )}`;
+              })
+              .join('\n');
+          })
+          .join('\n');
+      } else if (key === 'dependencies' || key === 'devDependencies') {
+        const contents = fs.readFileSync(pkgFile).toString();
+        return errors[key]
+          .map(packageName => {
+            const line = findLineInFile(contents, packageName);
+            const char = findCharacterInFile(contents, packageName);
+
+            return `  ${colors.dim(`${line}:${char}`)}  ${colors.red('error')}  ${colors.yellow(
+              packageName
+            )} is not used in code  ${colors.dim('check-dependencies-exist')}`;
+          })
+          .join('\n');
+      }
+    })
+    .join('\n\n');
```

Comment:
These changes update the output of the script to make an `eslint-stylish`-compatible output so that Github Actions creates an Annotation for any failures. Not strictly necessary, but it is nice when we don't have to comb over CI logs to see what and where failures occur.